cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -69,10 +69,11 @@ impl ArgAssigner for Args {
     fn assign(&mut self, arg: &AbiParam) -> ArgAction {
         let ty = arg.value_type;
 
-        // Check for a legal type.
-        // We don't support SIMD yet, so break all vectors down.
+        // Vectors should stay in vector registers TODO perhaps this should be conditional on some ISA feature (e.g. SSE) and if not present do a ValueConversion
         if ty.is_vector() {
-            return ValueConversion::VectorSplit.into();
+            let reg = FPR.unit(self.fpr_used);
+            self.fpr_used += 1;
+            return ArgumentLoc::Reg(reg).into();
```

Comment:
Someone should probably look at this to make sure I'm doing this right but, also, it seems like this should only happen when `enable_simd=true`--I don't see any reference to settings nearby that I can access, though.

Reply:
Pass in `settings` to `assign()`; pass in one big SettingsGroup if possible, otherwise pass in both a shared and ISA-specific SettingsGroup. Still need to split if SIMD is not enabled.